Client/Extensions/UIAlertControllerExtensions.swift

```diff
@@ -187,4 +195,106 @@ extension UIAlertController {
 
         return deleteAlert
     }
-}
+
+    /**
+     Builds the Alert view that asks if the users wants to add a third party search engine.
+
+     - parameter okayCallback: Okay option handler.
+
+     - returns: UIAlertController for asking the user to add a search engine
+     **/
+
+    class func addThirdPartySearchEngine(okayCallback: (UIAlertAction) -> Void) -> UIAlertController {
+        let alert = UIAlertController(
+            title: NSLocalizedString("AddTitle", tableName: "addSearchEngine", comment: "The title that asks the user to Add the search provider"),
+            message: NSLocalizedString("AddTitleMessage", tableName: "addSearchEngine", comment: "The message that asks the user to Add the search provider explaining where the search engine will appear"),
+            preferredStyle: UIAlertControllerStyle.Alert
+        )
+
+        let noOption = UIAlertAction(
+            title: NSLocalizedString("Cancel", tableName: "addSearchEngine", comment: "The cancel button if you do not want to add a search engine."),
+            style: UIAlertActionStyle.Cancel,
+            handler: nil
+        )
+
+        let okayOption = UIAlertAction(
+            title: NSLocalizedString("OK", tableName: "addSearchEngine", comment: "The confirmation button"),
+            style: UIAlertActionStyle.Default,
+            handler: okayCallback
+        )
+
+        alert.addAction(okayOption)
+        alert.addAction(noOption)
+        return alert
+    }
+
+    /**
+     Builds the Alert view that shows the user an error in case a search engine could not be added.
+
+     - returns: UIAlertController with an error dialog
+     **/
+
+    class func failedToAddThirdPartySearch() -> UIAlertController {
+        let alert = UIAlertController(
+            title: NSLocalizedString("FailedToAdd", tableName: "addSearchEngine", comment: "A title explaining that we failed to add a search engine"),
```

Comment:
Same here about using the real string (and everywhere that currently has a placeholder).